AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -160,9 +153,32 @@ - (void)_batchLayoutNodesFromContexts:(NSArray<ASIndexedNodeContext *> *)context
   }];
 }
 
+/**
+ * Perform measurement and layout of loaded or unloaded nodes based if they will be layed out on main thread or not
+ */
+- (void)_layoutNodes:(NSArray<ASCellNode *> *)nodes fromContexts:(NSArray<ASIndexedNodeContext *> *)contexts ofKind:(NSString *)kind
+{
+  ASDisplayNodeAssertTrue(nodes.count == contexts.count);
+  
+  if (_dataSource == nil) {
+    return;
+  }
+  
+  BOOL isMainThread = ASDisplayNodeThreadIsMain();
+  NSInteger idx = 0;
+  for (ASIndexedNodeContext *context in contexts) {
+    ASCellNode *node = nodes[idx];
+    // Only nodes that are loaded should be layout on the main thread
+    if (node.isNodeLoaded == isMainThread) {
```

Comment:
We should expand this comment. This code is quite creative, but for that reason it is just slightly difficult to read. It is not immediately clear that we intend to perform layout on unloaded nodes when we are not on the main thread.  A comment could make this more clear.

The comment should probably say that for any given layout pass that occurs, this method will be called at least twice, once on the main thread and the background, to result in complete coverage of both loaded and unloaded nodes